src/local-mode.ts

```diff
@@ -56,51 +102,150 @@ export class LocalModeManager {
 
 		if (!localMode) {
 			// Remove any leftovers if necessary
-			await this.removeLocalModeArtifacts();
+			await this.handleLocalModeStateChange(false);
```

Comment:
Found that this piece does not play very nice atm.

@CameronDiver would you give me more information on the use cases it handles?
Meanwhile, I'm going to make sure we update the snapshot after each new target state application, so that supervisor restart works correctly.

Reply:
Addressed in the new commit - we remove the stored snapshot after successful cleanup.